src/main/scala/castalia/StubService.scala

```diff
@@ -1,43 +1,64 @@
 package castalia
 
+import akka.actor.ActorSystem
 import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
 
-/**
-  * Created by Jens Kat on 27-11-2015.
-  */
-trait StubService extends BaseService {
+class StubService(theStubsByEndpoints: StubConfigsByEndpoint)(implicit val system: ActorSystem) extends Routes {
+  protected val serviceName = "StubRoutes"
 
-  protected val serviceName = "StubService"
+  protected def stubsByEndpoints: StubConfigsByEndpoint = theStubsByEndpoints
```

Comment:
I don't think this assignment is necessary. we can use `theStubsByEndpoints` directly. Also defs can also be implemented by vals (also in constructor arguments)